File: javascript/javascript-exercises/calculator/calculator.js
function add (a, b) {
	return (a + b);
}

function subtract (a, b) {
	return (a - b);
}

function sum (array) {
	return array.reduce((a, b) => a + b, 0);
}

function multiply (array) {
	return array.reduce((a, b) => a * b, 1);
}

function power(...array) {
	return array.reduce((a, b) => Math.pow(a, b));
	
}

function factorial(n) {
	return (n != 1 && n != 0) ? n * factorial(n - 1) : 1;
}

module.exports = {
	add,
	subtract,
	sum,
	multiply,
    power,
	factorial
}

File: ruby/learn_ruby/02_calculator/calculator.rb
#write your code here
def add(a, b)
    a + b
end

def subtract(a, b)
    a - b
end

def sum(array)
    array.sum()
end

def multiply(*args)
    args.reduce(:*)
end

def power(a, b)
    a**b
end

def factorial(n)
    if n==0
        1
    else
        n*factorial(n-1)
    end
end